flake.nix

```diff
@@ -102,17 +104,26 @@
             (import ./modules {
               inherit extraSpecialArgs home-manager-path pkgs;
               config.imports = modules;
+              config.build.arch =
+                nixpkgs.lib.strings.removeSuffix "-linux" system;
```

Comment:
In the throwIf block above, it is mentioned that system is deprecated and will be removed soon (tm). Maybe don't set a default value for pkgs, then you can just use pkgs.system

Reply:
> Maybe don't set a default value for pkgs, then you can just use pkgs.system

OK, I'm confused on how would that work.

nixpkgs-bootstrap aside, there are two more avenues we use and suggest to pass nixpkgs when using flakes:

1. through .follows (see minimal template), and for this one we can't use pkgs.system, since then we just don't know it,
2. through .pkgs (see advanced template), but I don't understand how would it determine one either.

Reply:
I was proposing to make `pkgs` (instantiated nixpkgs instance) a mandatory argument instead of `system` to be consistent with home-manager. I am also fine `system` as argument but then it should not trigger a warning when used.

Reply:
I think I understand now (45a3da6). Went the "pkgs is mandatory, system errors out" route.

I can make it work, but every time I need to pick between how do I do it, I'm about as good as a coin toss.

Reply:
Yeah true, both ways are fine. I agree on the coin toss regarding which direction to go :D